testing/MLDB-174-regression.py

```diff
@@ -0,0 +1,224 @@
+            }
+        }
+      
+        # start by training only with the pH and 'fixed acidity' features, specified
+        # excplicitely 
+        rez = mldb.put('/v1/procedures/wine_trainer', config)
+
+        scorerDetails = mldb.get("/v1/functions/winer_scorer_0/details").json()
+        mldb.log(scorerDetails)
+        usedFeatures = [x["feature"] for x in scorerDetails["model"]["params"]["features"]]
+        mldb.log(usedFeatures)
+        self.assertEqual(usedFeatures, ["pH", "fixed acidity"])
+
+        ## now try by using all features
+        config["params"]["trainingData"] = """
+                    select
+                    {* EXCLUDING(quality)} as features,
+                    quality as label
+                    from wine_red
+            """
+        rez = mldb.put('/v1/procedures/wine_trainer', config)
+
+        scorerDetails = mldb.get("/v1/functions/winer_scorer_0/details").json()
+        mldb.log(scorerDetails)
+        usedFeatures = [x["feature"] for x in scorerDetails["model"]["params"]["features"]]
+        mldb.log(usedFeatures)
+        self.assertGreater(len(usedFeatures), 2)
+
+    def test_wine_quality_merged_regression(self):
+        ## now create a merged dataset and train on that
+        # the problem here is that there are duplicated rowNames
+        mldb.put('/v1/procedures/column_adder', {
+            "type": "transform",
+            "params": {
+                "inputData": """
+
+                    SELECT * FROM merge(
+                        (
+                            SELECT *, 'red' as color
+                            FROM wine_red
+                        ),
+                        (
+                            SELECT *, 'white' as color
+                            FROM wine_white                  
+                        )               
+                    )
+                """,
+                "outputDataset": "wine_full",
+                "runOnCreation": True
+            }
+        })
+        
+        config = {
+            "type": "classifier.experiment",
+            "params": {
+                "trainingData": """
+                    select
+                    {* EXCLUDING(quality)} as features,
+                    quality as label
+                    from wine_full
+                """,
+                "datasetFolds": [
+                        {
+                            "training_where": "rowHash() % 2 = 0", 
+                            "testing_where": "rowHash() % 2 = 1"
+                        }
+                    ],
+                "experimentName": "winer",
+                "modelFileUrlPattern": "file://tmp/MLDB-174-wine.cls",
+                "algorithm": "glz",
+                "configuration": self.glz_conf,
+                "mode": "regression",
+                "runOnCreation": True
+            }
+        }
+
+        # this should fail because we check for dupes
+        with self.assertRaises(mldb_wrapper.ResponseException) as re:
```

Comment:
Consider using the regex assert to check that your new message has the expected format and content.